File: install_packages.R
# Install all R packages required for this project
# Run this script in R or RStudio to set up your R environment

cat("Installing R packages for Meme Stock Detection project...\n")
cat(paste(rep("=", 60), collapse = ""), "\n")

packages <- c(
  "tidyverse",
  "lubridate",
  "purrr",
  "tidymodels",
  "slider",
  "torch"
)

for (pkg in packages) {
  if (!require(pkg, character.only = TRUE, quietly = TRUE)) {
    cat(paste0("\nInstalling ", pkg, "...\n"))
    install.packages(pkg, dependencies = TRUE)
    cat(paste0("✓ ", pkg, " installed successfully\n"))
  } else {
    cat(paste0("✓ ", pkg, " already installed\n"))
  }
}

cat("\n", paste(rep("=", 60), collapse = ""), "\n")
cat("All R packages installed successfully!\n")
cat(paste(rep("=", 60), collapse = ""), "\n")

File: install_packages.py
"""
Install all Python packages required for this project.
Run this script to set up your Python environment.
"""

import subprocess
import sys

def install_packages():
    """Install all required Python packages."""
    packages = [
        'yfinance>=0.2.28',
        'pandas>=2.0.0',
        'numpy>=1.24.0',
        'requests>=2.31.0',
        'tqdm>=4.65.0',
    ]

    print("Installing Python packages for Meme Stock Detection project...")
    print("=" * 60)

    for package in packages:
        print(f"\nInstalling {package}...")
        try:
            subprocess.check_call([sys.executable, "-m", "pip", "install", package])
            print(f"✓ {package} installed successfully")
        except subprocess.CalledProcessError as e:
            print(f"✗ Failed to install {package}: {e}")
            return False

    print("\n" + "=" * 60)
    print("All Python packages installed successfully!")
    print("=" * 60)
    return True

if __name__ == "__main__":
    success = install_packages()
    sys.exit(0 if success else 1)